cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php

```diff
@@ -705,6 +705,28 @@ public function filter_image_block_pre_render( $block, $source_block ) {
 		return $block;
 	}
 
+
+	/**
+	 * Add filters for Rest API handling.
+	 */
+	public function init_rest_filters() {
+		// Gutenberg compatibility.
+		add_filter( 'rest_prepare_attachment', array( $this, 'filter_attachment_for_rest' ) );
+		$types  = get_post_types_by_support( 'editor' );
+		$filter = $this;
+		array_map(
```

Comment:
@DavidCramer 

Since we are refactoring this, I'd like you to consider changing this to a loop. After reading your comments and @dugajean, and after reading more about this, I think this is not a use case for `array_map`. In [PHP](https://www.php.net/manual/en/function.array-map.php) documentation for this, you can see that:

> array_map() returns an array containing the results of applying the callback to the corresponding index of array1 (and ... if more arrays are provided) used as arguments for the callback. The number of parameters that the callback function accepts should match the number of arrays passed to array_map().

This code is not leveraging any return from this function, and as stated before, it involves some unneeded overhead.

Again, I'm coming back to it only because we are refactoring, and it feels like the right opportunity.